File: samples/Go/entropy.go
package main

import (
    "fmt"
    "math"
)

const s = "1223334444"

func main() {
    m := map[rune]float64{}
    for _, r := range s {
        m[r]++
    }
    hm := 0.
    for _, c := range m {
        hm += c * math.Log2(c)
    }
    const l = float64(len(s))
    fmt.Println(math.Log2(l) - hm/l)
}

File: samples/Rust/entropy.rs
// works for Rust 0.9
fn entropy(s: &str) -> f32 {
	let mut entropy: f32 = 0.0;
	let mut histogram = [0, ..256];
	let len = s.len();

	for i in range(0, len) { histogram[s[i]] += 1; }
	for i in range(0, 256) {
		if histogram[i] > 0 {
			let ratio = (histogram[i] as f32 / len as f32) as f32;
			entropy -= (ratio * log2(ratio)) as f32;
		}
	}

	entropy
}
